app/code/Magento/Catalog/Block/Product/ListProduct.php

```diff
@@ -82,12 +89,14 @@ public function __construct(
         Resolver $layerResolver,
         CategoryRepositoryInterface $categoryRepository,
         Data $urlHelper,
+        CollectionLoader $collectionLoader,
```

Comment:
New injections should come last in the construct and should also default to null so as to not break backwards compatibility. More information and examples can be found at http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#adding-a-constructor-parameter